app/src/main/java/asd/org/ahorcado/activities/InputActivity.java

```diff
@@ -3,29 +3,47 @@
  */
 package asd.org.ahorcado.activities;
 
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
 import android.view.View;
 import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
 import android.widget.TextView;
+import android.widget.Toast;
 
 import asd.org.ahorcado.R;
 import asd.org.ahorcado.exceptions.MatchLostException;
 import asd.org.ahorcado.models.Match;
 
 public class InputActivity extends AppCompatActivity {
 
+    private static final int INTERVAL = 2000; //2 seconds
+
     private Match match = new Match();
```

Comment:
You should create AbstractMatch type, and then instantiate the Match class. Also I recommend put this one in a different method, and call it through the constructor.